cmd/maya-exporter/app/collector/zvol/metrics.go

```diff
@@ -65,224 +69,338 @@ type fields struct {
 }
 
 // newMetrics initializes fields of the metrics and returns its instance
-func newListMetrics() listMetrics {
-	return listMetrics{
-
-		used: prometheus.NewGaugeVec(
-			prometheus.GaugeOpts{
-				Namespace: "openebs",
-				Name:      "used_size",
-				Help:      "Used size of pool and volume",
-			},
-			[]string{"name"},
-		),
-
-		available: prometheus.NewGaugeVec(
-			prometheus.GaugeOpts{
-				Namespace: "openebs",
-				Name:      "available_size",
-				Help:      "Available size of pool and volume",
-			},
-			[]string{"name"},
-		),
-
-		zfsListParseErrorCounter: prometheus.NewGauge(
-			prometheus.GaugeOpts{
-				Namespace: "openebs",
-				Name:      "zfs_list_parse_error",
-				Help:      "Total no of zfs list parse errors",
-			},
-		),
-
-		zfsListCommandErrorCounter: prometheus.NewGauge(
-			prometheus.GaugeOpts{
-				Namespace: "openebs",
-				Name:      "zfs_list_command_error",
-				Help:      "Total no of zfs command errors",
-			},
-		),
-
-		zfsListRequestRejectCounter: prometheus.NewGauge(
-			prometheus.GaugeOpts{
-				Namespace: "openebs",
-				Name:      "zfs_list_request_reject_count",
-				Help:      "Total no of rejected requests of zfs list",
-			},
-		),
-	}
+func newListMetrics() *listMetrics {
+	return new(listMetrics)
+}
+
+func (l *listMetrics) withUsedSize() *listMetrics {
+	l.used = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "openebs",
+			Name:      "used_size",
```

Comment:
@utkarshmani1997 there is one metric for getting used capacity of pool 
```
Name:      "used_pool_capacity",
Help:      "Capacity used by pool",
```
and here used_size what is the diffrence between these two ? Can it be more descriptive like used_size to something more meaningful like used_pool_size or used_vol_size.